src/app/auth/permission.service.ts

```diff
@@ -0,0 +1,63 @@
+import { Injectable } from '@angular/core';
+import { JwtHelperService } from '@auth0/angular-jwt';
+
+export type Permission = {
+  exp: number,
+  resource_set_id: string,
+  resource_set_name: string,
+  scopes: Array<string>
+};
+
+@Injectable()
+export class PermissionService {
+  jwtHelper: JwtHelperService;
+  constructor() {
+    this.jwtHelper = new JwtHelperService();
+  }
+
+  /**
+   * Returns all the scopes a user has for a specific resource.
+   * @param resourceId ID of a specific resource such as a Space
+   */
+  getAllScopes(resourceId: string): Array<string> {
+    const permissions = this.getPermission(resourceId);
+    return permissions ? permissions.scopes : [];
+  }
+
+  /**
+   * Checks if a user has a specific scope for a resource.
+   * @param resourceId ID of a specific resource such as a Space
+   * @param scope the scope you want to check for. Ex - `can edit`
+   */
+  checkScope(resourceId: string, scope: string): boolean {
+    const permissions = this.getPermission(resourceId);
+    return permissions ? permissions.scopes.includes(scope) : false;
+  }
+
+  /**
+   * Returns the permission for a specific resource.
+   * @param resourceId ID of a specific resource such as a Space
+   */
+  getPermission(resourceId: string): Permission {
```

Comment:
We really need to start using strict nulls so that we can have better type checking. Right now this can return `undefined` but that's not clear from the return type.
This is something we will improve with the monorepo linting / tsconfig.